### ISSUE TITLE
Is there a way to exclude specific pages from production build

### ISSUE THREAD

**Issue by millette** [zeit/next.js#8974]
I know this was discussed in #2332 but I have a slightly different use case which I'd like to share.

# Feature request

Ignore certain pages under pages/ at build time.

## Is your feature request related to a problem? Please describe.

I'm using the same MDX pages (content) in two projects. One handles MDX at runtime and let's me catch malformed MDX/JSX at runtime. The other project uses the MDX files directly at buildtime, but now it fails on a page I was using to demonstrate error handling.

## Describe the solution you'd like

Could there be a `next.config.js` key, like `pageExtensions` but instead to prevent certain pages (by regex or simple array) from being built? Maybe `ignorePages`?

## Describe alternatives you've considered

Neither the [HOC approach](https://github.com/zeit/next.js/tree/canary/examples/with-higher-order-component) nor the [404 trick](https://github.com/zeit/next.js/issues/746) would help in the case of (known) malformed pages.

## Additional context

Note that I'm not looking to put otherwise useful files under `pages/` but I want to handle the case of known malformed files.

Back to my two-project example, you could think of one as the Admin and the other as the Public site. Files from Admin are automatically copied to Public using dat, but it would apply to any kind of sync.

**Comment by sideroad** [comment]
+1

**Comment by Timer** [comment]
This has been heavily debated in previous issues which can be found via search!

At present time, we're not exploring adding an option to remove specific pages from the production build.

tl;dr if they aren't pages, remove them from the `pages` directory.

You may be able to find more help for your unique use case on Spectrum!

**Comment by Lpaydat** [comment]
I just pass by but wonder about this too since one of my page is `.eslintrc.json` (it contains the local rules so it is not possible to move it to another dir).

It also shows when running `yarn next build` as a page too.

If I try to access it, e.g. at `http://localhost:3000/api/.eslintrc`, it will render `Internal Server Error` on the screen and this log on the server-side.

```
TypeError: resolver is not a function
    at Object.apiResolver (/home/<name>/Workspace/<project>/node_modules/next/dist/next-server/server/api-utils.js:42:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async DevServer.handleApiRequest (/home/<name>/Workspace/<project>/node_modules/next/dist/next-server/server/next-server.js:446:9)
    at async Object.fn (/home/<name>/Workspace/<project>/node_modules/next/dist/next-server/server/next-server.js:367:37)
    at async Router.execute (/home/<name>/Workspace/<project>/node_modules/next/dist/next-server/server/router.js:42:32)
    at async DevServer.run (/home/<name>/Workspace/<project>/node_modules/next/dist/next-server/server/next-server.js:487:29)
```

Maybe this is not a big deal but it may be good if we can exclude the hidden pages from the production build.

**Comment by kud** [comment]
I'm in a case that it sounds legit to me that my component is in `pages` without being seen as a page itself: 

![image](https://user-images.githubusercontent.com/655838/82388995-40417100-9a3b-11ea-851c-38c02dfcbea5.png)

**Comment by kud** [comment]
Workaround for the moment:

![image](https://user-images.githubusercontent.com/655838/82389764-446e8e00-9a3d-11ea-9975-3921608d5efb.png)

**Comment by jimisaacs** [comment]
In my case, I need to build the site twice for authenticated reasons. On one hand, there's unathed pages, and on another hand there's authed pages. It's extremely difficult to manage this with a single export without making them entirely different sites.

**Comment by jimisaacs** [comment]
Meaning, no matter what I do with `exportPathMap` or any other setting, knob, config value I've turned, the export always includes chunks/js assets related to the pages I am did not actually export.

**Comment by mmmulani** [comment]
wrote up how this could be added to Next.js in this discussion: https://github.com/vercel/next.js/discussions/29514 please chime in with feedback/support!

**Comment by lifedup** [comment]
I use getStaticProps to exclude static pages from builds.  For example, In my pages directory, I have a folder called admin.  Every page under that directory has the below getStaticProps.  

During the build and export, the admin directory is skipped.  I do not know if this will work for SSR pages.

```
export const getStaticProps: GetStaticProps = async () => {
  if (process.env.NODE_ENV === 'production') {
    return { notFound: true };
  }
  return { props: {} };
};
```

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
